src/main/webapp/script.js

```diff
@@ -1,88 +1,185 @@
+function loadDashboardPage() {
   $(document).ready(function() {
     $('#navbar').load('navbar.html', function() {
       $('#navbarDashboardSection').removeClass('d-none');
 
-      setupNavBarProfileSection();
-      loadSearchValue();
-      loadSortValue();
-      getDashboardItems();
+      initializeNavBarProfileSection();
+      initializeSearchInput();
+      initializeSortSelector();
+      initializeDashboard();
     });
 
     $('#itemSubmissionDiv').load('item-submission-dialog.html');
   });
 }
 
 /**
- * Loads the stored value used for the search input and adds a
- * callback to load the entertainment items when the user types a key.
+ * Enables access to the profile if the user is logged in by adding a "Profile"
+ * link to the navbar, if the user is not logged in then it adds a link to
+ * login.
  */
-function loadSearchValue() {
-  const searchInput = $('#searchValue');
+function initializeNavBarProfileSection() {
+  fetch('/login')
+      .then((response) => response.json())
+      .then((isUserLoggedIn) => {
+        const profileLinks = $('#profileLinks');
+
+        if (isUserLoggedIn) {
+          profileLinks.append($(
+              '<a class="nav-link text-light" href="/ProfilePage.html">Profile</a>'));
+        } else {
+          profileLinks.append($(
+              '<a class="nav-link text-light" href="/LoginPage.html">Login</a>'));
+        }
+      })
+      .catch((error) => {
+        console.log('failed to fetch login status: ' + error);
+      });
+}
+
+/**
+ * Loads the stored value used for the search input from sessionStorage
+ */
+function initializeSearchInput() {
   const searchVal = sessionStorage.getItem('searchVal');
 
   if (searchVal !== null) {
-    searchInput.val(searchVal);
+    $('#searchValue').val(searchVal);
   }
-
-  searchInput.on('input', function() {
-    sessionStorage.setItem('searchVal', $(this).val());
-    getDashboardItems();
-  });
 }
 
 /**
- * Loads the stored values for the sort type of the sort selector and adds a
- * callback to load the entertainment items when the selector changes value.
+ * Loads the stored values for the sort type of the sort selector from
+ * sessionStorage
  */
-function loadSortValue() {
-  const sortSelector = $('#sortType');
+function initializeSortSelector() {
   const sortVal = sessionStorage.getItem('sortVal');
 
   if (sortVal !== null) {
-    sortSelector.val(sortVal);
+    $('#sortType').val(sortVal);
   }
+}
 
-  sortSelector.change(function() {
+/**
+ * Fetches for the entertainment items from DashboardServlet and the favorite
+ * item ids from FavoriteItemServlet to initialize the dashboard with items that
+ * have correct like button state.
+ */
+async function initializeDashboard() {
+  Promise
+      .allSettled([
+        fetch(
+            '/dashboard?searchValue=' + $('#searchValue').val() +
+            '&sortType=' + $('#sortType').val()),
+        fetch('/favorite-item')
+      ])
+      .then(async (results) => {
```

Comment:
just curious, do you need the `async` keyword here?

Reply:
Yes the async function is required because inside this method I am making calls to .json() which is an async function.